flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/planner/delegation/hive/HiveParserRexNodeConverter.java

```diff
@@ -761,8 +761,7 @@ private List<RexNode> rewriteCaseChildren(
                             .makeNullLiteral(
                                     newChildRexNodeLst
                                             .get(newChildRexNodeLst.size() - 1)
-                                            .getType()
-                                            .getSqlTypeName()));
+                                            .getType()));
```

Comment:
`org.apache.calcite.rex.RexBuilder#makeNullLiteral` is depecated at https://issues.apache.org/jira/browse/CALCITE-1621 with recommendation to use `#makeNullLiteral(RelDataType)`
```java
  // CHECKSTYLE: IGNORE 1
  /** @deprecated Use {@link #makeNullLiteral(RelDataType)} */
  @Deprecated // to be removed before 2.0
  public RexNode makeNullLiteral(SqlTypeName typeName) {
    return makeNullLiteral(typeFactory.createSqlType(typeName));
  }
```